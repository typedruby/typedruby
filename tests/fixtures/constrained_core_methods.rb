# @typedruby

def array_compact1 => nil
  reveal_type [1, nil, 3].compact
  nil
end

def array_compact2 => nil
  reveal_type [nil, 2, 3].compact
  nil
end

def array_compact_no_nil => nil
  reveal_type [1, 2, 3].compact
  nil
end

def array_to_h => nil
  reveal_type [["one", 1], ["two", 2]].to_h
  nil
end

def array_to_h_err => :any
  [["one", 1], 2].to_h
  nil
end
