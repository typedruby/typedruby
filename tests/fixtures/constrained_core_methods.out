error: Could not match types:

        @ __ROOT__/tests/fixtures/constrained_core_methods.rb:1
      1 |  def array_compact => nil
                                ^^^ NilClass, with:
        @ (builtin stdlib):809
    809 |    def compact[NonNullType; ~NonNullType = ElementType] => [NonNullType]; end
                                                                     ^^^^^^^^^^^^^ Array::[Integer]

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/constrained_core_methods.rb:1
      1 |  def array_compact => nil
                                ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/constrained_core_methods.rb:2
      2 |    [1, nil, 3].compact
             ^^^^^^^^^^^^^^^^^^^ Array::[Integer]

error: Could not match types:

        @ (builtin stdlib):809
    809 |    def compact[NonNullType; ~NonNullType = ElementType] => [NonNullType]; end
                                      ^^^^^^^^^^^^ NilClass | t2, with:
        @ __ROOT__/tests/fixtures/constrained_core_methods.rb:6
      6 |    [1, 2, 3].compact
              ^ Integer

        - arising from an attempt to match:

        @ (builtin stdlib):809
    809 |    def compact[NonNullType; ~NonNullType = ElementType] => [NonNullType]; end
                                      ^^^^^^^^^^^^ NilClass | t2, with:
        @ (builtin stdlib):809
    809 |    def compact[NonNullType; ~NonNullType = ElementType] => [NonNullType]; end
                                                     ^^^^^^^^^^^ Integer
        @ (builtin stdlib):809
    809 |    def compact[NonNullType; ~NonNullType = ElementType] => [NonNullType]; end
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^ in this expression

error: Could not match types:

        @ __ROOT__/tests/fixtures/constrained_core_methods.rb:9
      9 |  def array_to_h => nil
                             ^^^ NilClass, with:
        @ (builtin stdlib):827
    827 |    def to_h[K, V; ElementType = [K, V]] => { K => V }; end
                                                     ^^^^^^^^^^ Hash::[String, Integer]

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/constrained_core_methods.rb:9
      9 |  def array_to_h => nil
                             ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/constrained_core_methods.rb:10
     10 |    [["one", 1], ["two", 2]].to_h
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Hash::[String, Integer]

error: Could not match types:

        @ __ROOT__/tests/fixtures/constrained_core_methods.rb:14
     14 |    [["one", 1], 2].to_h
             ^^^^^^^^^^^^^^^ [String, Integer] | Integer, with:
        @ (builtin stdlib):827
    827 |    def to_h[K, V; ElementType = [K, V]] => { K => V }; end
                                          ^^^^^^ [t3, t4]

        - arising from an attempt to match:

        @ (builtin stdlib):827
    827 |    def to_h[K, V; ElementType = [K, V]] => { K => V }; end
                            ^^^^^^^^^^^ [String, Integer] | Integer, with:
        @ (builtin stdlib):827
    827 |    def to_h[K, V; ElementType = [K, V]] => { K => V }; end
                                          ^^^^^^ [t3, t4]
        @ (builtin stdlib):827
    827 |    def to_h[K, V; ElementType = [K, V]] => { K => V }; end
                            ^^^^^^^^^^^^^^^^^^^^ in this expression
